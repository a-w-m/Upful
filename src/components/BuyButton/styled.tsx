import styled from "styled-components"
import device from "../layout/mediaQuery"

interface ButtonProps{
  "data-item-max-quantity"?: number
}

export const Button = styled.button<ButtonProps>`
  cursor: pointer;
  font-size: 1.4rem;
  font-weight: 500;
  letter-spacing: .1rem;
  background-color: var(--main-bg);
  color: var(--header-nav-link-hover);
  line-height: 1.4rem;
  border: 0.1rem solid var(--main-brown);
  padding: 1.5rem 3.2rem;
  transition: all 0.15 ease;
  border-radius: 0rem;
  opacity: ${props=> props["data-item-max-quantity"]=== undefined || props["data-item-max-quantity"] > 0 ? 1 : 0.3};
  user-select: ${props=> props["data-item-max-quantity"]=== undefined || props["data-item-max-quantity"] > 0 ?"auto" : "none"};
  width: 100%;
  text-transform: uppercase;


 @media ${device.laptop}{
   margin-bottom: 0;
 }
 
`
