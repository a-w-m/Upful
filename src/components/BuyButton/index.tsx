import React from "react"
import { Button } from "./styled"
import { P } from "../../interfaces/index"
import { createCustomOptionsProps, createOptionsString } from "../../helpers"

interface Props {
  "data-item-id": String
  "data-item-price": String
  "data-item-url": String
  "data-item-name": String
  "data-item-description": String
  "data-item-image": String
  "data-item-weight"?: number
  "data-item-max-quantity": number
  productOptions: P.ProductOption[]
  optionsSelected: P.State["optionsSelected"]
}

const BuyButton: React.FC<Props> = (props: Props) => {
  const customOptions = createCustomOptionsProps(
    props.productOptions,
    props.optionsSelected
  )

  return (
    <Button
      className="snipcart-add-item"
      data-item-id={props["data-item-id"]}
      data-item-price={props["data-item-price"]}
      data-item-url={props["data-item-url"]}
      data-item-name={props["data-item-name"]}
      data-item-description={props["data-item-description"]}
      data-item-image={props["data-item-image"]}
      data-item-weight={props["data-item-weight"]}
      data-item-max-quantity={props["data-item-max-quantity"]}
      {...customOptions}
      disabled={props["data-item-max-quantity"] > 0 ? false : true}
    >
      {props["data-item-max-quantity"] > 0 ? "Add to Cart" : "Sold Out"}
    </Button>
  )
}

export default BuyButton
