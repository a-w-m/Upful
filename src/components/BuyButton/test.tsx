import React from 'react'
import {render} from '@testing-library/react'
import BuyButton from "."

describe("BuyButton Component", ()=>{
    test("matches snapshot", ()=>{
        const {container} = render( <BuyButton
            className ='snipcart-add-item'
            data-item-id= ""
            data-item-price= ""
            data-item-name= ""
            data-item-description=""
            data-item-image=""
            data-item-url=""
            data-item-custom1-name=""
            data-item-custom1-options=""
            data-item-custom1-value="">
       </BuyButton>)
        expect(container.firstChild).toMatchSnapshot()

    })
})