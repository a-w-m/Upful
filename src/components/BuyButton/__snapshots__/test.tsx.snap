// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyButton Component matches snapshot 1`] = `
<button
  class="sc-bczRLJ bZchrX snipcart-add-item"
  data-item-description=""
  data-item-id=""
  data-item-image=""
  data-item-max-quantity="1"
  data-item-name=""
  data-item-price="10"
  data-item-url=""
>
  Add to Cart
</button>
`;
